packages/scheduler/lib/scheduler.ts

```diff
@@ -203,11 +204,11 @@ export class Scheduler {
      * @example
      * const cancelled = await scheduler.cancel({ taskId: '00000000-0000-0000-0000-000000000000' });
      */
-    public async cancel(cancelBy: { taskId: string } | { scheduleId: string }): Promise<Result<Task>> {
+    public async cancel(cancelBy: { taskId: string; reason: string } | { scheduleId: string; reason: string }): Promise<Result<Task>> {
```

Comment:
we could also make the second param a `JsonValue` and allow any kind of object. Let me know what you think.